File: data_structures/Stack/Stack.py



class Stack:
    def __init__(self):
        self.stack = []

    def push(self, val):
        self.stack.append(val)

    def pop(self):
        if len(self.stack) > 0:
            self.stack.pop(-1)

    def peek(self):
        if len(self.stack) > 0:
            return self.stack[-1]
        

    def getData(self):
        if len(self.stack) > 0:
            print(self.stack)
        else:
            print("Stack is empty !")



def main():
    newStack = Stack()

    print("Pushing !")

    newStack.push(1)
    newStack.getData()
    newStack.push(2)
    newStack.getData()
    newStack.push(3)
    newStack.getData()
    newStack.push(4)
    newStack.getData()
    newStack.push(5)
    newStack.getData()

    print("Popping !")

    newStack.getData()
    newStack.pop()
    newStack.getData()
    newStack.pop()
    newStack.getData()
    newStack.pop()
    newStack.getData()
    newStack.pop()
    newStack.getData()
    newStack.pop()
    newStack.getData()


if __name__ == '__main__':
    main()

File: data_structures/Stack/Stack.js



class Stack {
  constructor () {
    this.stack = [];
  }
  
  push (val) {
    this.stack.push(val);
  }

  pop (val) {
    if (this.stack.length > 0) {
       this.stack.pop();
    } else {
      return null;
    }
  }

  getData () {
    if (this.stack.length !== 0) {
      console.log(this.stack);
    } else {
      console.log("Stack is empty !");
    }
  }
}

let newStack = new Stack()

console.log("Pushing !");

newStack.push(1);
newStack.getData();
newStack.push(2);
newStack.getData();
newStack.push(3);
newStack.getData();
newStack.push(4);
newStack.getData();
newStack.push(5);
newStack.getData();

console.log("Popping !")

newStack.getData();
newStack.pop();
newStack.getData();
newStack.pop();
newStack.getData();
newStack.pop();
newStack.getData();
newStack.pop();
newStack.getData();
newStack.pop();
newStack.getData();
newStack.pop();

